### PR TITLE
CQ-4325360 : Improvements for workflow-migrator (Platform Independence)

### PR DESCRIPTION
**Pull request by dichaudhary**
## Description

- The changes will fix the use cases where the tool fails for windows machine 
- Also the failing test cases are fixed
- The Path.of takes care of platform independence wherever possible
- We have to diifferentiate between the file system paths and jcr paths.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by dichaudhary** [comment]
The PR has passed the failing test cases on windows before, I have also tested the changes using the projects for which it was not generating the reports earlier which it did after the changes.

cc: @ManasMaji  @ukgaurav

File: src/main/java/com/adobe/skyline/migration/MigrationConstants.java [hunk 1]
```diff
@@ -195,11 +195,11 @@
...
-        put("dam/process_subasset", buildWorkflowLauncher("/content/dam(/.*/)(subassets)(/.*/)renditions/original", Arrays.asList("jcr:content/jcr:mimeType!=video/.*")));
-        put("dam/update_asset", buildWorkflowLauncher("/content/dam(/((?!/subassets).)*/)renditions/original", new ArrayList<>()));
-        put("dam/update_from_lightbox", buildWorkflowLauncher("/var/lightbox", new ArrayList<>()));
+        put("batch-thumbnails", buildWorkflowLauncher("/var/dam/pending-thumbs(/.*)", Arrays.asList("paths!=")));
+        put("dynamic-media-encode-video", buildWorkflowLauncher("/content/dam(/.*/)renditions/original", Arrays.asList("jcr:content/jcr:mimeType==video/.*")));
+        put("process_subasset", buildWorkflowLauncher("/content/dam(/.*/)(subassets)(/.*/)renditions/original", Arrays.asList("jcr:content/jcr:mimeType!=video/.*")));
+        put("update_asset", buildWorkflowLauncher("/content/dam(/((?!/subassets).)*/)renditions/original", new ArrayList<>()));
+        put("update_from_lightbox", buildWorkflowLauncher("/var/lightbox", new ArrayList<>()));
```
Comment (ireasor) [inline src/main/java/com/adobe/skyline/migration/MigrationConstants.java:202]: The OOTB workflow launchers are under a dam subnode.  I'm surprise that this change didn't introduce bugs.

**Review by ireasor — COMMENTED** [review]
The Path.of() change looks good and is actually a little cleaner than the String concatenation that I had originally put in.  Thanks for that.  I had one concern about the updated workflow paths in the MigrationConstants, though.

File: src/main/java/com/adobe/skyline/migration/MigrationConstants.java (hunk 1 above)
Comment (dichaudhary) [reply src/main/java/com/adobe/skyline/migration/MigrationConstants.java:202]: Hi @ireasor, thanks for reviewing the PR, actually the changes in https://github.com/adobe/aem-cloud-migration/pull/14/files#diff-35f15394733556cf28ff40d8b83e3dc6575c8d023b7aaae04d88a53d436337e7R129 will only bring the actual name of the workflow model and the reason why the above changes are done!

Comment (ireasor) [reply src/main/java/com/adobe/skyline/migration/MigrationConstants.java:202]: Ah, makes sense.  Thanks!

**Review by ireasor — APPROVED** [review]

**Review by ManasMaji — APPROVED** [review]

**Pull request merged**